SubBox/app/src/main/java/com/joelimyx/subbox/MainList/MainActivity.java

```diff
@@ -0,0 +1,317 @@
+import com.joelimyx.subbox.R;
+import com.joelimyx.subbox.dbassethelper.DBAssetHelper;
+import com.joelimyx.subbox.dbassethelper.SubBoxHelper;
+import com.joelimyx.subbox.history.HistoryActivity;
+import com.joelimyx.subbox.history.HistoryFragment;
+
+import java.text.DateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.IllegalFormatCodePointException;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class MainActivity extends AppCompatActivity implements SubBoxAdapter.OnItemSelectedListener {
+    @BindView(R.id.recyclerview) RecyclerView mRecyclerView;
+    @BindView(R.id.progressbar) ProgressBar mProgressBar;
+    ArrayAdapterSearchView mArrayAdapterSearchView;
+    ArrayAdapter<String> mArrayAdapter;
+    FrameLayout container;
+    SubBoxAdapter mAdapter;
+    private boolean mTwoPane;
+    public static final int DETAIL_REQUEST_CODE = 1;
+    public static final int CHECKOUT_REQUEST_CODE = 2;
+    public static final int HISTORY_REQUEST_CODE = 3;
+    private MainAsyncTask mTask;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        ButterKnife.bind(this);
+
+        container = (FrameLayout) findViewById(R.id.detail_or_checkout_container);
+        mTwoPane = container!=null;
+
+        DBAssetHelper dbSetup = new DBAssetHelper(MainActivity.this);
+        dbSetup.getWritableDatabase();
+
+        mRecyclerView.setLayoutManager(new GridLayoutManager(MainActivity.this,2, LinearLayoutManager.VERTICAL,false));
+        mTask = new MainAsyncTask();
+        mTask.execute(new Object());
+
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+    }
+
+    //--------------------------------------------------------------------------------------------------------------------
+    //Make sure the detail_or_checkout_container stay the same as it as it(detail or checkout) was before rotation
+    //--------------------------------------------------------------------------------------------------------------------
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if(mTwoPane) {
+            if (requestCode == DETAIL_REQUEST_CODE) {
+                if (resultCode == RESULT_OK) {
+                    int id = data.getIntExtra("id", -1);
+                    DetailFragment detailFragment = DetailFragment.newInstance(id);
+                    getSupportFragmentManager().beginTransaction().replace(R.id.detail_or_checkout_container, detailFragment).commit();
+                }
+            }else if(requestCode == CHECKOUT_REQUEST_CODE){
+                if (resultCode == RESULT_OK){
+                    CheckOutFragment checkOutFragment = new CheckOutFragment();
```

Comment:
Use `CheckOutFragment.newInstance()` rather than the constructor here. I know you don't have any parameters to pass to `newInstance()` so it doesn't matter now, but in the future you might have something to pass to the fragment that way.